libs/oidc/common/src/lib/services/client-metadata/client-metadata.service.ts

```diff
@@ -236,6 +256,21 @@ export class ClientMetadataService {
     return redirectUris;
   }
 
+  // BackchannelLogoutUri functions
+  public async createBackchannelLogoutUri(_backchannelLogoutUris: string[]) {
+    const backchannelLogoutUris: BackchannelLogoutUri[] = [];
+
+    _backchannelLogoutUris.map((value, i) => {
+      const backchannelLogoutUri: Partial<BackchannelLogoutUri> = {
+        url: value
+      };
+      const logoutUri = this.backchannelUriRepo.create(backchannelLogoutUri);
+      backchannelLogoutUris.push(logoutUri);
+    });
+
+    return backchannelLogoutUris;
+  }
+
```

Comment:
Nothing about this block is async. Can cause a race condition since 271 will not necessary return after the  `_backchannelLogoutUris` are completely mapped.

Wrong use of `Array.map()`.